database/service_db.go

```diff
@@ -77,39 +82,66 @@ func (d *LevelDBServiceDB) All() ([]*service.Service, error) {
 }
 
 // Delete deletes service from database.
-func (d *LevelDBServiceDB) Delete(id string) error {
+func (d *LevelDBServiceDB) Delete(idOrAlias string) error {
+	id, err := d.aliases.Get([]byte(idOrAlias), nil)
+	if err != nil && err != leveldb.ErrNotFound {
+		return err
+	}
+	if string(id) != "" { // has alias, deleting it...
+		idOrAlias = string(id)
+		if err := d.aliases.Delete([]byte(idOrAlias), nil); err != nil {
+			return err
+		}
+	}
 	return d.db.Delete([]byte(id), nil)
 }
 
 // Get retrives service from database.
-func (d *LevelDBServiceDB) Get(id string) (*service.Service, error) {
-	b, err := d.db.Get([]byte(id), nil)
+func (d *LevelDBServiceDB) Get(idOrAlias string) (*service.Service, error) {
+	id, err := d.aliases.Get([]byte(idOrAlias), nil)
+	if err != nil && err != leveldb.ErrNotFound {
+		return nil, err
+	}
+	if string(id) != "" {
+		idOrAlias = string(id)
+	}
+	b, err := d.db.Get([]byte(idOrAlias), nil)
 	if err != nil {
 		if err == leveldb.ErrNotFound {
-			return nil, &ErrNotFound{ID: id}
+			return nil, &ErrNotFound{ID: idOrAlias}
 		}
 		return nil, err
 	}
 
-	return d.unmarshal(id, b)
+	return d.unmarshal(idOrAlias, b)
 }
 
 // Save stores service in database.
 func (d *LevelDBServiceDB) Save(s *service.Service) error {
-	if s.ID == "" {
-		return errors.New("database: can't save service without id")
+	if s.ID == "" || s.Alias == "" {
```

Comment:
Maybe we should always require an ID and _alias_ can stay as optional. Because we always save services by using their ID as key.

Reply:
alias should be optional for the user but needs to be required internally